File: app/src/main/java/com/example/c0766552_w2020_mad3125_fp/ui/SplashActivity.java
package com.example.c0766552_w2020_mad3125_fp.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.example.c0766552_w2020_mad3125_fp.R;
import com.example.c0766552_w2020_mad3125_fp.util.DataStorage;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        DataStorage.getInstance().loadDetails();

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent hint = new Intent(SplashActivity.this, LoginActivity.class);
                startActivity(hint);
                finish();
            }
        },5000);
    }
}

File: app/src/main/java/com/example/c0766552_w2020_mad3125_fp/ui/SplashActivity.kt
package com.example.c0766552_w2020_mad3125_fp.ui

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.example.c0766552_w2020_mad3125_fp.R
import com.example.c0766552_w2020_mad3125_fp.util.DataStorage

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        DataStorage.getInstance().loadDetails()
        Handler().postDelayed({
            val hint = Intent(this@SplashActivity, LoginActivity::class.java)
            startActivity(hint)
            finish()
        }, 5000)
    }
}